tests/console/commands/test_lock.py

```diff
@@ -204,6 +222,45 @@ def test_lock_no_update_path_dependencies(
     assert {p.name for p in packages} == {"quix", "sampleproject"}
 
 
+def test_lock_directory_dependency_folder_does_not_exist(
+    command_tester_factory: CommandTesterFactory,
+    poetry_with_missing_directory_dependency: Poetry,
+):
+    locker = Locker(
+        lock=poetry_with_missing_directory_dependency.pyproject.file.path.parent
+        / "poetry.lock",
+        local_config=poetry_with_missing_directory_dependency.locker._local_config,
+    )
+    poetry_with_missing_directory_dependency.set_locker(locker)
+
+    tester = command_tester_factory(
+        "lock", poetry=poetry_with_missing_directory_dependency
+    )
+    with pytest.raises(
+        ValueError, match=r"Source directory .* for missing does not exist"
+    ):
+        tester.execute("")
+
+
+def test_lock_directory_dependency_deleted_from_pyproject(
+    command_tester_factory: CommandTesterFactory,
+    poetry_with_deleted_directory_dependency_from_pyproject_and_filesystem: Poetry,
+):
+    poetry = poetry_with_deleted_directory_dependency_from_pyproject_and_filesystem
+    locker = Locker(
+        lock=poetry.pyproject.file.path.parent / "poetry.lock",
+        local_config=poetry.locker._local_config,
+    )
+    poetry.set_locker(locker)
+
+    tester = command_tester_factory("lock", poetry=poetry)
+    tester.execute("")
```

Comment:
Not really. Here's the current matrix of what works/doesn't.

## poetry lock --no-update
### Current
|            | in pyproject | not in pyproject but in lockfile |
|------------|---------------------------|----------------------------------|
| exists     |             ✅             |                 ✅                |
| not exists |             ❌             |                 ❌                |
### Desired
|            | in pyproject | not in pyproject but in lockfile |
|------------|---------------------------|----------------------------------|
| exists     |             ✅             |                 ✅                |
| not exists |             ❌             |                 ✅                |

## poetry lock
### Current
|            | in pyproject | not in pyproject but in lockfile |
|------------|---------------------------|----------------------------------|
| exists     |             ✅             |                 ✅                |
| not exists |             ❌             |                 ✅                |
### Desired
|            | in pyproject | not in pyproject but in lockfile |
|------------|---------------------------|----------------------------------|
| exists     |             ✅             |                 ✅                |
| not exists |             ❌             |                 ✅                |


This test is target at making sure the `not exists/in pyproject` (❌) still fails with `poetry lock`.

We do need a test to make sure the `not exists/not in pyproject but in lockfile` (before ❌ now ✅) works as well as `not exists/in pyproject` (❌) still fails with `lock --no-update`. I guess I was hesitant to add it in this PR because it seemed unrelated, but now that this PR is just "update the tests after https://github.com/python-poetry/poetry-core/pull/507#pullrequestreview-1169630432 gets merged" it seems appropriate. I'll add that.

Reply:
Actually never mind: you are totally right. What I wrote above still applies, just not to this test case!